File: src/testp1/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package testp1;

/**
 *
 * @author usuario
 */
public class Weapon {
    private float power;
    private int uses;
    
    Weapon(float _power, int _uses){
        power=_power;
        uses=_uses;
    }
    
    public float attack(){
        if (uses>0){
            uses--;
            return power;
        }
        else{
            return 0;
        }
    }
    
    public String toString(){
        return "W["+power+","+uses+"]";
    }
    
    public boolean discard(){
        Dice d = new Dice();
        return d.discardElement(uses);
    }
}

File: RUBY/weapon.rb
# frozen_string_literal: true
module Irrgarten
class Weapon
  def initialize(power, uses)
    @power=power
    @uses=uses
  end

  def attack
    if @uses>0
      uses-=1
      @power
    else
      0
    end
  end

  def to_s
    "W[ #{@power}, #{@uses}]"
  end

  def discard
    Dice.discardElement(@uses)
  end
end
end